impeller/entity/entity_unittests.cc

```diff
@@ -1736,5 +1742,149 @@ TEST_P(EntityTest, SrgbToLinearFilter) {
   ASSERT_TRUE(OpenPlaygroundHere(callback));
 }
 
+TEST_P(EntityTest, TTTBlendColor) {
+  std::vector<BlendMode> blend_modes = {
+      BlendMode::kClear,
+      BlendMode::kSource,
+      BlendMode::kDestination,
+      BlendMode::kSourceOver,
+      BlendMode::kDestinationOver,
+      BlendMode::kSourceIn,
+      BlendMode::kDestinationIn,
+      BlendMode::kSourceOut,
+      BlendMode::kDestinationOut,
+      BlendMode::kSourceATop,
+      BlendMode::kDestinationATop,
+      BlendMode::kXor,
+      BlendMode::kPlus,
+      BlendMode::kModulate,
+  };
+
+  auto check_func = [blend_modes](Color& src, Color& dst,
+                                  std::vector<Color> expected_colors) {
+    FML_LOG(ERROR) << "Source " << src << " Dst " << dst;
+    for (unsigned i = 0; i < blend_modes.size(); ++i) {
+      auto blend_color = Color::BlendColor(src, dst, blend_modes[i]);
+      auto expected_color = expected_colors[i];
+      ASSERT_EQ(blend_color, expected_color);
+      FML_LOG(ERROR) << "blend_mode: " << i;
+    }
+  };
+
+  auto generator_expected_colors = [](Color& src,
```

Comment:
I would not use a closure like this to generate the expected colors. A frequent mistake made with this sorts of tests, is that someone writes some code (with a bug), copies and pastes the implementation into the test generator (with the same bug) and asserts that they are the same - which they will be, but still wrong.

Instead each test case should have a specified color that a developer could easily cross reference with an external source

Reply:
done